### ISSUE TITLE
[HOLD for payment 2024-07-10] [HOLD for payment 2024-07-02] [$125] Web - Markdown - Quote line becomes invisible when hovering over it

### ISSUE THREAD

**Issue by kbecciv** [Expensify/App#41901]
If you haven’t already, check out our [contributing guidelines](https://github.com/Expensify/ReactNativeChat/blob/main/contributingGuides/CONTRIBUTING.md) for onboarding and email contributors@expensify.com to request to join our Slack channel!
___

**Version Number:** 1.4.72-0
**Reproducible in staging?:** y
**Reproducible in production?:** y
**Issue reported by:** Applause - Internal Team

## Action Performed:
1. Navigate to staging.new.expensify.com
2. Create an expense with a quote mark down description
3. Go to expense details page and hover over the description

## Expected Result:
The quote line has different color from the hover effect and becomes visible while hovering

## Actual Result:
The quote line becomes invisible because of having the same color as the hover effect color

## Workaround:
n/a

## Platforms:

Which of our officially supported platforms is this issue occurring on?
- [ ] Android: Native
- [ ] Android: mWeb Chrome
- [ ] iOS: Native
- [ ] iOS: mWeb Safari
- [x] MacOS: Chrome / Safari
- [ ] MacOS: Desktop

## Screenshots/Videos

Add any screenshot/video evidence

https://github.com/Expensify/App/assets/93399543/9bf39cfa-3ec6-4170-b52f-6c4c1a26bf01

[View all open jobs on GitHub](https://github.com/Expensify/App/issues?q=is%3Aopen+is%3Aissue+label%3A%22Help+Wanted%22)

<details><summary>Upwork Automation - Do Not Edit</summary>
    <ul>
        <li>Upwork Job URL: https://www.upwork.com/jobs/~019cfcf6cb0e02864c</li>
        <li>Upwork Job ID: 1788623606867111936</li>
        <li>Last Price Increase: 2024-06-10</li>
<li>Automatic offers: </li>
<ul>
<li>rojiphil | Reviewer | 0</li>
</ul></li>
    </ul>
</details>

<details><summary>Issue Owner</summary>Current Issue Owner: @CortneyOfstad</details>

**Comment by kbecciv** [comment]
@CortneyOfstad FYI I haven't added the External label as I wasn't 100% sure about this issue. Please take a look and add the label if you agree it's a bug and can be handled by external contributors.

**Comment by CortneyOfstad** [comment]
Was able to recreate without issue

https://github.com/Expensify/App/assets/10926327/fd808e42-45db-4819-b9a3-4b7c50bf16c7

**Comment by isogit123** [comment]
## Proposal
### Please re-state the problem that we are trying to solve in this issue.
The Markdown quote line disappears when hovering over it.

### What is the root cause of that problem?
The quote line and the hover background have the same colour as obvious below:

https://github.com/Expensify/App/blob/807c945ef8edabe86cc62f16c866aa754db8750b/src/styles/index.ts#L175-L176

### What changes do you think we should make in order to solve the problem?
Define a separate colour to be used for the blockquote.
In the ThemeColors type, add a property for example by typing `blockquote: Color;`.
After that, in both the `darkTheme` and `lightTheme` objects, define a color to be used for the blockquote for example `blockquote: colors.productDark600,` for the `darkTheme` and `blockquote: colors.productLight600,` for the `lightTheme`.
Lastly, replace `theme.border` with `theme.blockquote` in:

https://github.com/Expensify/App/blob/807c945ef8edabe86cc62f16c866aa754db8750b/src/styles/index.ts#L175-L176

and

https://github.com/Expensify/App/blob/807c945ef8edabe86cc62f16c866aa754db8750b/src/styles/index.ts#L3221-L3222

and 

https://github.com/Expensify/App/blob/807c945ef8edabe86cc62f16c866aa754db8750b/src/hooks/useMarkdownStyle.ts#L3221-L3222

Here is a POC after applying the solution:


https://github.com/Expensify/App/assets/34800355/42123e21-e3ed-4eb8-a9ff-638210743cd1

Contributor details
Your Expensify account email: [aboutabl6@gmail.com](mailto:aboutabl6@gmail.com)
Upwork Profile Link: https://www.upwork.com/fl/islama33

**Comment by CortneyOfstad** [comment]
@rojiphil we have a proposal [here](https://github.com/Expensify/App/issues/41901#issuecomment-2104899229). Can you provide feedback by EOD today please?

**Comment by rojiphil** [comment]
Thanks for your proposals. Both proposals point out the correct root cause i.e. quote line and hover background color are the same. The solution is also straightforward i.e. change the blockquote color to something else in consultation with the design team. 
Since @dragnoir was the first to identify the root cause and the solution, his [proposal](https://github.com/Expensify/App/issues/41901#issuecomment-2102643312) LGTM. 
Also, since the solution did not involve any logical implementation and only involved changing color it is fine not to give a detailed solution.
🎀👀🎀 C+ reviewed

**Comment by MonilBhavsar** [comment]
Updating the price since it is minor issue and fix

**Comment by CortneyOfstad** [comment]
> Updating the price since it is minor issue and fix cc @CortneyOfstad

Thanks @MonilBhavsar!

**Comment by CortneyOfstad** [comment]
The merge freeze is planned to end today 🤞

**Comment by MonilBhavsar** [comment]
The PR was reverted because it caused a design bug

**Comment by MonilBhavsar** [comment]
![image](https://github.com/Expensify/App/assets/32012005/cc26d1d2-8b82-4a2a-974e-a3f1bad24a96)
Hovering and active tabs style were affected

https://expensify.slack.com/archives/C049HHMV9SM/p1717057643081609

**Comment by CortneyOfstad** [comment]
@rojiphil Can you please complete the checklist before tomorrow so there is no delay in payment?

Also @rojiphil and @dragnoir — I've sent you both offers in Upwork, please let me know once you accept and I will get those paid right away tomorrow. Thanks!

**Comment by MonilBhavsar** [comment]
Shouldn't we fix this again? The PR was actually reverted

**Comment by CortneyOfstad** [comment]
@MonilBhavsar I don't see a revert on the PR. Can you point me out where you see that?

**Comment by MonilBhavsar** [comment]
Yes, the issue is not fixed on staging


https://github.com/Expensify/App/assets/32012005/0ff42cc0-8b1e-4bd1-900e-0289af5e2a41

**Comment by MonilBhavsar** [comment]
Revert PR https://github.com/Expensify/App/pull/42828
https://github.com/Expensify/App/issues/41901#issuecomment-2139383663

**Comment by MonilBhavsar** [comment]
Yes, we should making PR that fixes this issue and does not affect other components in the App, and cause a regression

**Comment by CortneyOfstad** [comment]
Gotcha, thanks @MonilBhavsar!

@rojiphil for the correcting PR, do you have an ETA as to when you could have that drafted/ready for review?

**Comment by MonilBhavsar** [comment]
Have we thought about altering the quote line color when hovering?
It is dark by default, on hovering it becomes light and on hovering out, it becomes dark again.

**Comment by MonilBhavsar** [comment]
Thanks!

**Comment by rojiphil** [comment]
> I will check with the design team throw Slack is you allow me

@dragnoir Not sure if you initiated a slack discussion already but I think we need inputs from the design team on your comment [here](https://github.com/Expensify/App/issues/41901#issuecomment-2140477947)  i.e. change hover/active color or change quote line color. 
Is this correct?

**Comment by rojiphil** [comment]
@dragnoir 
> design team already asked for the update here [#42313 (comment)](https://github.com/Expensify/App/pull/42313#issuecomment-2115955608)

Ok. Got it. Thanks. 

> Have we thought about altering the quote line color when hovering?
> It is dark by default, on hovering it becomes light and on hovering out, it becomes dark again.

I also like this approach. So, the next step is to check with the design team as you mentioned.

**Comment by shawnborton** [comment]
I think where your PR was wrong was that it changed the background color of :active menu items in the LHN. We don't want to do that, we are only changing the hover color for when you hover over a row.

**Comment by CortneyOfstad** [comment]
Going to be adjusting the assignments as Rachid is no longer working on the project. Still looking for proposals!

**Comment by rojiphil** [comment]
> I see that [@isogit123](https://github.com/isogit123) had a similar proposal [here](https://github.com/Expensify/App/issues/41901#issuecomment-2104899229). Do we want to assign them to keep this rolling? I'll let you re-reviewing the proposal and see if any adjustments need to be made or feedback provided.

Sure @CortneyOfstad. Let’s give it a try. 
@isogit123 Can you please update your [proposal](https://github.com/Expensify/App/issues/41901#issuecomment-2104899229) to reflect [this approach](https://github.com/Expensify/App/issues/41901#issuecomment-2150573206)?

**Comment by isogit123** [comment]
@rojiphil Updating the blockquote colour on hover is hard and tricky as the colour is applied globally over the application.
The cause of the issue that happened after merging Dragnoir's PR was the last change in the following commit:
https://github.com/Expensify/App/commit/74618dfbcd3c403bba0e5514cbe3386dafc90aad#diff-097e3d461183958f84757dc8be87e3f24b862654235a9b3f59f75ae74a738e1f
that is shown in the following image:

![image](https://github.com/Expensify/App/assets/34800355/ea8ea466-2a80-48bd-8442-5e3dfe6e3364)
which changed the buttons' active style. The following image shows the result of applying that change without changing the blockquote colour, which shows the issue:

![problem](https://github.com/Expensify/App/assets/34800355/af707b59-2f24-483e-9c98-7b5eccea1507)

So, I suggest following my [proposal](https://github.com/Expensify/App/issues/41901#issuecomment-2104899229) which is a similar approach by changing the blockquote colour, but without doing the change mentioned above in the button's active style. The following two images show that the blockquote is visible on hover and that there is no issue in the hovering and active tabs styles.

![Screenshot from 2024-06-13 18-15-34](https://github.com/Expensify/App/assets/34800355/56e58593-e160-4c27-9268-e944b50b2dd5)

![Screenshot from 2024-06-13 18-15-55](https://github.com/Expensify/App/assets/34800355/e061507f-58a5-4fe1-a5d9-0496c8f6690f)

**Comment by isogit123** [comment]
Contributor details:
Your Expensify account email: [aboutabl6@gmail.com](mailto:aboutabl6@gmail.com)
Upwork Profile Link: https://www.upwork.com/fl/islama33

**Comment by shawnborton** [comment]
Hmm this screenshot looks wrong though:
![image](https://github.com/Expensify/App/assets/2319350/22e6d725-36a0-4b0c-a78a-4100c49f4fba)

Maybe it exists in the product, but I think :hovering over any push row or option row should use our rowHover color, which does not collide with our border color.

**Comment by isogit123** [comment]
@shawnborton I have been able to achieve what you suggested. Here is an updated proposal to do it:

## Proposal
### Please re-state the problem that we are trying to solve in this issue.
The Markdown quote line disappears when hovering over it.

### What is the root cause of that problem?
The `MenuItem` hover background has the wrong colour, which is the same as the quote line colour.

### What changes do you think we should make in order to solve the problem?
Correct the `MenuItem` hover background colour to use the rowHover colour. By adding the following:
`isHovered && !pressed && styles.hoveredComponentBG`
after line 464 in the following code snippet:

https://github.com/Expensify/App/blob/807c945ef8edabe86cc62f16c866aa754db8750b/src/components/MenuItem.tsx#L440-L465

Here is a POC after applying the solution:

https://github.com/Expensify/App/assets/34800355/edf9bb37-97d6-4003-a9de-42f5d43f0f9a

The video also shows that there is no issue in the hovering and active tabs styles.

**Contributor details:**
Your Expensify account email: [aboutabl6@gmail.com](mailto:aboutabl6@gmail.com)
Upwork Profile Link: https://www.upwork.com/fl/islama33

**Comment by rojiphil** [comment]
> Correct the `MenuItem` hover background colour to use the rowHover colour. By adding the following:
> `isHovered && !pressed && styles.hoveredComponentBG`
> after line 464 in the following code snippet:

Yes. That would work. Thanks for the updated proposal.
@isogit123 [proposal](https://github.com/Expensify/App/issues/41901#issuecomment-2166940890) LGTM
🎀👀🎀 C+ reviewed

**Comment by MonilBhavsar** [comment]
@CortneyOfstad I think we can reset the price to $250 because it didn't turn out to be one liner fix and there was some back and forth discussion to conclude the solution.

**Comment by isogit123** [comment]
Thank you for assigning me. I applied to the job.

Contributor details:
Your Expensify account email: aboutabl6@gmail.com
Upwork Profile Link: https://www.upwork.com/fl/islama33

**Comment by rojiphil** [comment]
> Thank you for assigning me. I applied to the job.

@isogit123 If this is the first time you are raising PR here, please go through the contributing guidelines [here](https://github.com/Expensify/App/blob/main/contributingGuides/CONTRIBUTING.md#begin-coding-your-solution-in-a-pull-request) for steps.

**Comment by MonilBhavsar** [comment]
@isogit123 thank you! Please let us know when you expect to have a PR ready for review

**Comment by isogit123** [comment]
I expect it by today or tomorrow.

**Comment by MonilBhavsar** [comment]
Okay, thanks for the update. Feel free to ask any questions you have here or #expensify-open-source slack channel

**Comment by isogit123** [comment]
PR is ready.
https://github.com/Expensify/App/pull/43969

**Comment by isogit123** [comment]
I created a new PR to merge with the latest changes without conflicts.

https://github.com/Expensify/App/pull/43987

**Comment by isogit123** [comment]
I think the PR caused a regression here https://github.com/Expensify/App/issues/44211
I can't verify that because I don't see the "Book Travel" option after clicking the FAB. Is there something that I'm missing here?
![image](https://github.com/Expensify/App/assets/34800355/846ffa06-cba2-486c-abfe-2bc6f4efbdbc)

**Comment by rojiphil** [comment]
> I can't verify that because I don't see the "Book Travel" option after clicking the FAB. Is there something that I'm missing here?

@isogit123 Yea. We missed considering the `interactive` hover case. `Book Travel` is beta feature. You can return true [here](https://github.com/Expensify/App/blob/425804aa005c6299ff0f59ff596764c429e19abd/src/libs/Permissions.ts#L7) and you will be able to see this. Can you please bring up a PR for this?

**Comment by CortneyOfstad** [comment]
@rojiphil can you please have the checklist done by EOD so there is no delay in issuing payment? Thank you!

**Comment by CortneyOfstad** [comment]
@rojiphil any update on the checklist?

Also, @isogit123 I am having trouble finding your Upwork profile. Can you please link it here so I can send a proposal to you for payment? Thank you!

**Comment by isogit123** [comment]
@CortneyOfstad https://www.upwork.com/freelancers/~011b9b21b42a676c99

**Comment by CortneyOfstad** [comment]
Thank you @isogit123! For some reason the original job post was not letting me search for you so I had to manually send you an offer which can be found [here](https://www.upwork.com/nx/wm/offer/102962562). Let me know once you accept and I can get that paid ASAP. Thank you!

**Comment by isogit123** [comment]
@CortneyOfstad I accepted it. Thank you.

**Comment by CortneyOfstad** [comment]
Also @rojiphil it looks like you still have not accepted your offer as well in Upwork. Please accept this as soon as you can so I can get that paid. Thanks!

**Comment by rojiphil** [comment]
* [x]  [@rojiphil] The PR that introduced the bug has been identified. Link to the PR: [Offending PR](https://github.com/Expensify/App/pull/13748)
* [x]  [@rojiphil] The offending PR has been commented on, pointing out the bug it caused and why, so the author and reviewers can learn from the mistake. Link to comment: [Added comment](https://github.com/Expensify/App/pull/13748/files#r1667010883)
* [x]  [@rojiphil] A discussion in #expensify-bugs has been started about whether any other steps should be taken (e.g. updating the PR review checklist) in order to catch this type of bug sooner. Link to discussion: Not Required. Existing checklist is good enough to capture such issues.
* [x]  [@rojiphil] Determine if we should create a regression test for this bug. :  Since the changes are related to styling in menu item which is widely used, this is prone to regression. Let’s add regression test case for this. 
* [x]  [@rojiphil] If we decide to create a regression test for the bug, please propose the [regression test](https://github.com/Expensify/App/blob/main/contributingGuides/REGRESSION_TEST_BEST_PRACTICES.md) steps to ensure the same bug will not reach production again. 
1. Create an expense with a quote mark down description.
2. Go to expense details page and hover over the description.
3. Verify that the quote line is visible while hovering.

**Comment by rojiphil** [comment]
> Also @rojiphil it looks like you still have not accepted your offer as well in Upwork. Please accept this as soon as you can so I can get that paid. Thanks!

@CortneyOfstad Completed BZ checklist. I could not accept the offer as it expired. Can you please send a fresh offer for this? Thanks.

**Comment by CortneyOfstad** [comment]
Sorry, was OoO — @rojiphil sent you an updated offer. Let me know once you accept and I can get that paid ASAP tomorrow. Thank you!

**Comment by rojiphil** [comment]
> sent you an updated offer. Let me know once you accept and I can get that paid ASAP tomorrow.

@CortneyOfstad Offer Accepted. Thanks

**Comment by CortneyOfstad** [comment]
@isogit123 I am having trouble finding you in Upwork to send you the proposal. Can you share your Upwork profile and I'll get the proposal sent over ASAP to get it paid. Thank you!

**Comment by CortneyOfstad** [comment]
@isogit123 I don't see the offer — can you link me your Upwork profile?

**Comment by isogit123** [comment]
@CortneyOfstad https://www.upwork.com/freelancers/~011b9b21b42a676c99

**Comment by CortneyOfstad** [comment]
Thanks @isogit123! I checked Upwork and was able to find the proposal, but it is waiting for you to accept. Once that is done, I can get this paid for you ASAP. Thanks!

**Comment by isogit123** [comment]
@CortneyOfstad I received an offer of $125, and it was paid successfully as illustrated in the attached photo.
Now, I received a second one, Do I accept that too?![Screenshot_20240711_222146_Upwork Talent.jpg](https://github.com/user-attachments/assets/54f0b333-f1f7-49f4-9134-8145c8fab83b)

**Comment by CortneyOfstad** [comment]
That is so weird — there is no history of that in Upwork anywhere. However, if you were already paid, then no worries, I will retract the original offer. 

I am so sorry for the confusion here! Upwork is not being cooperative 😂

**Comment by CortneyOfstad** [comment]
Okay, now that it is sorted, moving forward with posting the Payment Summary

**Comment by CortneyOfstad** [comment]
## Payment Summary

@isogit123 — paid $125 via Upwork
@rojiphil — paid $125 via Upwork 

## Regression Test

https://github.com/Expensify/Expensify/issues/412157
